### PR TITLE
Support Rails 7.1

### PR DESCRIPTION
**Pull request by bforma**

### CODE REVIEW

**Comment by lvonk** [comment]
Can you add `7.1` to `.github/workflows/rspec.yml` ?

File: lib/sequent/support/database.rb [hunk 1]
```diff
@@ -53,7 +53,6 @@ def self.establish_connection(db_config, db_config_overrides = {})
           db_config = db_config.deep_merge(
             Sequent.configuration.primary_database_key => db_config_overrides,
           ).stringify_keys
-          ActiveRecord.legacy_connection_handling = false
```
Comment (lvonk) [inline lib/sequent/support/database.rb:56]: We might need to keep this if AR_VERSION is < 7.2 ? @sam-cw ?

Comment (bforma) [reply lib/sequent/support/database.rb:56]: Hmm yes, I got failing tests telling me to remove it.

Comment (bforma) [reply lib/sequent/support/database.rb:56]: It's fixed.

**Comment by lvonk** [comment]
Would you mind squashing them in a single commit 🙏🏼

**Comment by bforma** [comment]
:shipit:

**Pull request merged**